### PR TITLE
Add Carter

### PR DESCRIPTION
**Pull request by benaadams**
https://github.com/CarterCommunity/Carter

/cc @jchannon for approval

### CODE REVIEW

File: frameworks/CSharp/carter/Benchmarks/Benchmarks.csproj [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <OutputType>Exe</OutputType>
+    <NoDefaultLaunchSettingsFile>true</NoDefaultLaunchSettingsFile>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Carter" Version="3.6.0" />
```
Comment (jchannon) [inline frameworks/CSharp/carter/Benchmarks/Benchmarks.csproj]: Just published https://www.nuget.org/packages/Carter/3.7.0 with netcore21 support

File: frameworks/CSharp/carter/Benchmarks/Program.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+namespace Benchmarks
+{
+    using System.IO;
+    using Microsoft.AspNetCore.Hosting;
+    using Microsoft.Extensions.Configuration;
+
+    public class Program
+    {
+        public static void Main(string[] args)
```
Comment (jchannon) [inline frameworks/CSharp/carter/Benchmarks/Program.cs]: Should this be an async Main?

File: frameworks/CSharp/carter/Benchmarks/PlainModule.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+        public PlainModule() : base("plaintext")
+        {
+            Get("/", (req, res, routeData) =>
+            {
+                var payloadLength = _helloWorldPayload.Length;
+                res.StatusCode = 200;
+                res.ContentType = "text/plain";
+                res.ContentLength = payloadLength;
+                return res.Body.WriteAsync(_helloWorldPayload, 0, payloadLength);
```
[the remark this answers is not in the document]
Comment (jchannon) [reply frameworks/CSharp/carter/Benchmarks/PlainModule.cs]: I'm not fussed tbh, if this is better lets go with it ;)

**Comment by benaadams** [comment]
Committed async Main + update Carter

@jchannon do you want me to make the other changes? With them it does look a lot nicer, but will also run slower (e.g. chunked encoding for plaintext and always converting from string to bytes for every request vs already in bytes and content-length response). Pick one:

1. Idiomatic 
2. Raw speed

I'd go for 2. then have beautiful samples in your repo (as you do) - but you'd probably expect that 😉

**Comment by jchannon** [comment]
Yeah go for 2 😄

**Comment by jchannon** [comment]
Thanks for taking the time to do this 👍

File: frameworks/CSharp/carter/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Carter Tests on Linux
+This includes tests for plaintext and json serialization.
+
+## Infrastructure Software Versions
+
+**Language**
+
+* C# 7.0
```
Comment (jchannon) [inline frameworks/CSharp/carter/README.md]: Does this need to be 7.3?

File: frameworks/CSharp/carter/benchmark_config.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      "approach": "Realistic",
+      "classification": "Micro",
+      "database": "None",
+      "framework": "Carter",
+      "language": "C#",
+      "orm": "Raw",
+      "platform": ".NET",
+      "flavor": "CoreCLR",
+      "webserver": "Kestrel",
+      "os": "Linux",
```
Comment (jchannon) [inline frameworks/CSharp/carter/benchmark_config.json]: Do we want a Windows run too?

Comment (benaadams) [reply frameworks/CSharp/carter/benchmark_config.json]: TE isn't currently setup for Windows 😢

File: frameworks/CSharp/carter/README.md (hunk 4 above)
Comment (benaadams) [reply frameworks/CSharp/carter/README.md]: Well async main is only 7.0; I just set the project to 7.3 out of habit :)

**Comment by jchannon** [comment]
STATS!

When do we want them?

**Comment by nbrady-techempower** [comment]
image

**Pull request merged**